### PR TITLE
Remove oneDNN-specific attributes from matmul

### PR DESCRIPTION
**Pull request by Silv3S**
### PR types
Others

### PR changes
OPs

### Describe
In https://github.com/PaddlePaddle/Paddle/pull/49515 each fuse done for `matmul` and `matmul_v2` was converted to be executed as `fused_matmul`. Fusion logic was extracted to dedicated fused op and fused kernel.  As base `matmul` is now not fused with anything, all extra attributes can be deleted from OPMaker, .pbtxt file and unit tests. 

Changes done in this PR:
- Remove fusion logic from base `matmul` op and kernel,
- Remove placeholders for empty attributes in unit tests,
- Remove _test_matmul_mkldnn_op.py_ unit tests, which has extra attributes hardcoded in base operators. These test cases are covered in _test_onednn_matmul_transpose_reshape_fuse_pass.py_ and _test_onednn_reshape_transpose_matmul_fuse_pass.py_. Same for `matmul_v2`.

### CODE REVIEW

**Comment by Silv3S** [comment]
`reshape_transpose_matmul_mkldnn_fuse_pass` introduced 4 extra oneDNN-specific attributes for matmul op: `fused_reshape_X`, `fused_reshape_Y`, `fused_transpose_X`, `fused_transpose_Y`. Then in https://github.com/PaddlePaddle/Paddle/pull/30011 checkpoint with only one of these attributes (`fused_reshape_Y`) was added to op version registry.

Now, we are going to delete all oneDNN-specific attributes from base ops. Do you think we could just delete this checkpoint and restore matmul version 0 or new checkpoint should be added and matmul would be version 2 then? @YuanRisheng, @chenwhql please advice

**Review by jczaja — DISMISSED** [review]
LGTM

**Comment by chenwhql** [comment]
> `reshape_transpose_matmul_mkldnn_fuse_pass` introduced 4 extra oneDNN-specific attributes for matmul op: `fused_reshape_X`, `fused_reshape_Y`, `fused_transpose_X`, `fused_transpose_Y`. Then in #30011 checkpoint with only one of these attributes (`fused_reshape_Y`) was added to op version registry.
> 
> Now, we are going to delete all oneDNN-specific attributes from base ops. Do you think we could just delete this checkpoint and restore matmul version 0 or new checkpoint should be added and matmul would be version 2 then? @YuanRisheng, @chenwhql please advice

If these attributes are only used for mkldnn, and you are sure that the removal is no problem, I think we can just delete this checkpoint and restore matmul version 0, and matmul op is no longer in use, removing the extra attribute should have no other effect

File: paddle/phi/api/yaml/op_compat.yaml [hunk 1]
```diff
@@ -1076,9 +1076,7 @@
   outputs :
     out : Out
   extra :
-    attrs : [bool use_mkldnn = false, 'int[] fused_reshape_Out = {}', 'int[] fused_transpose_Out = {}',
-             str mkldnn_data_type = "float32", 'int[] fused_reshape_X = {}', 'int[] fused_reshape_Y = {}',
-             'int[] fused_transpose_X = {}', 'int[] fused_transpose_Y = {}']
+    attrs : [bool use_mkldnn = false, str mkldnn_data_type = "float32"]
```
Comment (wozna) [inline paddle/phi/api/yaml/op_compat.yaml:1237]: Is mkldnn_data_type attr still needed in matmul op?

Comment (Silv3S) [reply paddle/phi/api/yaml/op_compat.yaml:1237]: I think it's still needed for bf16 passes as we don't convert `matmul` to `fused matmul` during cpu_bf16_placement_pass.

There are still some other extra attributes in base `matmul` - for example int8 scales, but it's easier to gradually remove attributes in few PRs to catch any regression. This PR removes just 6 attributes which were used by `matmul+transpose+reshape` and `reshape+transpose+matmul` fuses.

**Review by jczaja — APPROVED** [review]

**Comment by yaomichael** [comment]
@YuanRisheng please help review, approve and merge

**Review by YuanRisheng — APPROVED** [review]

**Review by zyfncg — APPROVED** [review]

**Review by chenwhql — APPROVED** [review]

**Comment by Silv3S** [comment]
@zhangting2020 could you please approve removing decorator `@skip_check_grad_ci`? I deleted test cases with extra hardcoded attributes from _test_matmul_mkldnn_op.py_ as they are already covered in _test_onednn_matmul_transpose_reshape_fuse_pass.py_.

**Comment by Silv3S** [comment]
@XiaoguangHu01 could you please approve exceeding limit of 20 modified files?

**Review by zhangting2020 — APPROVED** [review]
LGTM.

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**